### ISSUE TITLE
Types not inferred through anonymous function idiom (IFFE)

### ISSUE THREAD

**Issue by blickly** [google/closure-compiler#306]
This issue was imported from Closure Compiler's previous home at http://closure-compiler.googlecode.com

The original discussion is archived at:
http://blickly.github.io/closure-compiler-issues/#822

**Comment by Dominator008** [comment]
Looks like this has been fixed. The following code does not give any error / warning:
```js
var namespace = {}

/** @constructor */
namespace.MyType1 = function() {};

(function() {
  /** @constructor */
  namespace.MyType2 = function() {};
})();

(function(ns) {
  /** @constructor */
  ns.MyType3 = function() {};
})(namespace);


/** @return {namespace.MyType1} */
function returnMyType1() {
  return new namespace.MyType1();
}

// "Bad type annotation"
/** @return {namespace.MyType2} */
function returnMyType2() {
  return new namespace.MyType2();
}

// "Bad type annotation"
/** @return {namespace.MyType3} */
function returnMyType3() {
  return new namespace.MyType3();
}
```
